crates/extensions/c8y_http_proxy/src/tests.rs

```diff
@@ -135,11 +135,11 @@ async fn retry_internal_id_on_expired_jwt() {
 
 #[tokio::test]
 async fn retry_internal_id_on_expired_jwt_with_mock() {
-    let device_id = "device-001";
+    let external_id = "device-001";
     let tmp_dir = "/tmp";
-    let external_id = "external-device-001";
+    let internal_id = "external-device-001";
 
-    let response = InternalIdResponse::new(device_id, external_id);
+    let response = InternalIdResponse::new(internal_id,external_id);
```

Comment:
This is really easy to mess-up, as this has been done for this test.

- The names are misleading: the device_id expected here in the internal device id (internal from the c8y perspective), while device_id usually means the device certificate CN in the context of thin-edge (external if from the c8y perspective).
- The types should not be Strings but specific types wrapping strings.

This will have to be done in a different PR.